### PR TITLE
Added some missing Standard java classpath functions/interfaces

### PR DESCRIPTION
**Pull request by lwahlmeier**
@d2chau Take a look at the ListIterator implementations and make sure I did that correctly.  It definitely needs a 2nd set of eyes.

### CODE REVIEW

File: classpath/java/util/ArrayList.java [hunk 1]
```diff
@@ -183,6 +183,20 @@ public void clear() {
...
+  
+  @Override
+  public boolean retainAll(Collection<?> c) {
+    boolean changed = false;
+    Iterator<T> iter = this.iterator();
+    while(iter.hasNext()) {
+      T item = iter.next();
+      if(!c.contains(item)) {
+        changed = true;
+        iter.remove();
```
Comment (d2chau) [inline classpath/java/util/ArrayList.java]: Should we be removing the iterator when still in a loop? If changed is true, would it be good to return changed from there and break from the loop?

File: classpath/java/util/Collections.java [hunk 2]
```diff
@@ -500,10 +514,41 @@ public boolean hasNext() {
...
+
+    @Override
+    public void add(T e) {
+      list.add(index, e);
+    }
+
+    @Override
+    public int previousIndex() {
+      if(hasPrevious()) {
+
```
Comment (d2chau) [inline classpath/java/util/Collections.java]: Is this intentionally empty?

File: classpath/java/io/OverrideablePrintStream.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  
+  public void write(int c) throws IOException {
+    if(override == null) {
+      super.write(c);
+    } else {
+      override.write(c);
+    }
+  }
+
+  public void write(byte[] buffer, int offset, int length) throws IOException {
```
Comment (d2chau) [inline classpath/java/io/OverrideablePrintStream.java]: Should this throw an IOException? I didn't see it in the docs. 
http://docs.oracle.com/javase/7/docs/api/java/io/PrintStream.html#write(byte[],%20int,%20int)

File: classpath/java/util/ArrayList.java (hunk 1 above)
Comment (lwahlmeier) [inline classpath/java/util/ArrayList.java]: The idea behind retainAll (i believe) is to only retain items in the collection backing collection that are in the passed in collection, so any/all items in the backing collection that are not in the passed in one need to be removed.  We do not technically need to set changed to true after each one, but its not worth a conditional each time.  We do however have to go through the whole list of items in the backing collection.

File: classpath/java/util/Collections.java (hunk 2 above)
Comment (lwahlmeier) [inline classpath/java/util/Collections.java]: nope, we should probably be returning index-1, or maybe index depending on how we are keeping track.  Feel free to fix it, or I will once I get a chance.

File: classpath/java/io/OverrideablePrintStream.java (hunk 3 above)
Comment (lwahlmeier) [inline classpath/java/io/OverrideablePrintStream.java]: You are right, it shouldnt.  Feel free to update it, or I will when I get a chance.

**Pull request merged**